vortex-array/src/arrays/fixed_size_list/vtable/operations.rs

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::ops::Range;
+
+use vortex_scalar::Scalar;
+
+use crate::arrays::{FixedSizeListArray, FixedSizeListVTable};
+use crate::vtable::{OperationsVTable, ValidityHelper};
+use crate::{ArrayRef, IntoArray};
+
+impl OperationsVTable<FixedSizeListVTable> for FixedSizeListVTable {
+    fn slice(array: &FixedSizeListArray, range: Range<usize>) -> ArrayRef {
+        let start = range.start;
+        let end = range.end;
+
+        debug_assert!(
+            start <= end && end <= array.len(),
+            "slice [{start}..{end}) out of bounds: then len is {}",
+            array.len()
+        );
+
+        let new_len = end - start;
+        let list_size = array.list_size() as usize;
+
+        FixedSizeListArray::new(
```

Comment:
here's an example of where we should use `new_unchecked` instead, since we're maintaining the invariant of slicing the elements child ourselves